File: src/miniprogram/api/member.js
import request from './index'

/**
 * 会员 API
 */
export const memberApi = {
  /**
   * 获取充值方案
   */
  getPlans() {
    return request({ url: '/member/plans' })
  },

  /**
   * 获取会员信息
   */
  getInfo() {
    return request({ url: '/member/info' })
  },

  /**
   * 发起充值
   */
  recharge(plan) {
    return request({ url: '/member/recharge', method: 'POST', data: { plan } })
  },

  /**
   * 充值记录
   */
  recharges(params) {
    return request({ url: '/member/recharges', data: params })
  }
}

File: src/server/router/member.go
package router

import (
	"yule-go/handler"
	"yule-go/repository"
	"yule-go/service"

	"github.com/gin-gonic/gin"
)

// RegisterMemberRoutes 注册会员相关路由（需要用户 JWT 认证）
func RegisterMemberRoutes(r *gin.RouterGroup) {
	rechargeRepo := repository.NewRechargeRepository()
	userRepo := repository.NewUserRepository()
	memberSvc := service.NewMemberService(rechargeRepo, userRepo)
	memberH := handler.NewMemberHandler(memberSvc)

	member := r.Group("/member")
	{
		member.GET("/plans", memberH.GetPlans)
		member.GET("/info", memberH.GetMemberInfo)
		member.POST("/recharge", memberH.CreateRecharge)
		member.GET("/recharges", memberH.ListRecharges)
	}
}

// RegisterMemberCallbackRoutes 注册充值回调路由（无需认证）
func RegisterMemberCallbackRoutes(r *gin.RouterGroup) {
	rechargeRepo := repository.NewRechargeRepository()
	userRepo := repository.NewUserRepository()
	memberSvc := service.NewMemberService(rechargeRepo, userRepo)
	memberH := handler.NewMemberHandler(memberSvc)

	r.POST("/member/recharge/callback", memberH.RechargeCallback)
}
